frame/evm/src/precompiles.rs

```diff
@@ -67,3 +70,108 @@ impl Precompiles for Tuple {
 		None
 	}
 }
+
+/// Linear gas cost
+fn linear_cost(len: usize, base: usize, word: usize) -> Option<usize> {
```

Comment:
Why isn't this directly returning `Result` with `Err(OutOfGas)`, instead of doing this at each call?